## Portfolio and Risk Management

This project is used as a tool to practice and apply methods I have learned about Portfolio and Risk Management. This project will be expanded on a rolling basis, mostly limited by how much time I have after school, etc... 
First we need to get some data.

In [2]:
## Imports

import matplotlib as plt
import pandas as pd
import numpy as np
import time
from datetime import datetime
import yfinance as yf

In [3]:
## Tickers part of the portfolio
tickers = ['TSLA', 'GME', 'AMC' ,'MSFT', 'GOOG']

portfolio = pd.read_csv('Data/Portfolio.csv', index_col = 0)

portfolio.columns = tickers

The next step would be to check for completness of the data and familiarize yourself with the structure of the data.

In [4]:
portfolio.head()

TSLA        GME        AMC       MSFT        GOOG
Date                                                              
2016-01-04  44.681999  21.848803  17.572334  49.591064  741.840027
2016-01-05  44.686001  22.203815  17.777107  49.817295  742.580017
2016-01-06  43.807999  21.895105  17.595087  48.912342  743.619995
2016-01-07  43.130001  21.956850  17.155214  47.211044  726.390015
2016-01-08  42.200001  21.895105  16.973190  47.355843  714.469971

In [5]:
portfolio.describe()

TSLA          GME          AMC         MSFT         GOOG
count  1521.000000  1521.000000  1521.000000  1521.000000  1521.000000
mean    221.454803    40.280898    16.201018   135.963729  1336.691101
std     291.042571    63.279808    10.567634    80.806940   612.532663
min      28.733999     2.800000     1.980000    44.449951   668.260010
25%      50.310001    10.100000     9.489752    67.510132   930.390015
50%      65.820000    15.945121    13.260000   107.105705  1140.989990
75%     286.152008    22.180662    22.130543   200.470337  1485.930054
max    1229.910034   347.510010    62.549999   343.109985  3014.179932

In [6]:
portfolio.isnull().values.any()

False

It also seems to be helpful to plot the plot prices as a function of time. However, since some of the assets have largely different prices the scale might be affected.

In [7]:
import plotly.express as px
import plotly.io as pio
import plotly.graph_objects as go

pio.renderers.default = "svg"

fig = px.line(portfolio)

fig.update_layout(legend_title = "Asset", title = "Asset Prices over Time")
fig.update_xaxes(title = "Date")
fig.update_yaxes(title = "Stock Prices (in USD)")
fig.update_layout(width = 950, height = 700, title_x = 0.5, template = "ggplot2")

fig.write_image("Images/Asset_Prices_Over_Time.png")

![title](Images/Asset_Prices_Over_Time.png)

As expected, the scale suffers. Let's plot the cummulative returns. This should increase legibility of the plot.


In [8]:
asset_returns = portfolio.pct_change().dropna()

asset_returns.head()

TSLA       GME       AMC      MSFT      GOOG
Date                                                        
2016-01-05  0.000090  0.016249  0.011653  0.004562  0.000998
2016-01-06 -0.019648 -0.013903 -0.010239 -0.018165  0.001400
2016-01-07 -0.015477  0.002820 -0.025000 -0.034783 -0.023170
2016-01-08 -0.021563 -0.002812 -0.010610  0.003067 -0.016410
2016-01-11 -0.014929  0.035601 -0.004021 -0.000573  0.002184

In [9]:
asset_returns_cum = ((1+asset_returns).cumprod()-1) * 100

asset_returns_cum.head()

TSLA       GME       AMC      MSFT      GOOG
Date                                                        
2016-01-05  0.008956  1.624862  1.165314  0.456192  0.099751
2016-01-06 -1.956046  0.211924  0.129481 -1.368638  0.239940
2016-01-07 -3.473430  0.494524 -2.373731 -4.799292 -2.082661
2016-01-08 -5.554806  0.211924 -3.409587 -4.507308 -3.689482
2016-01-11 -6.964772  3.779568 -3.798007 -4.562054 -3.479186

In [10]:
fig = px.line(asset_returns_cum)

fig.update_layout(legend_title = "Asset", title = "Cummulative Return over Time")
fig.update_xaxes(title = "Date")
fig.update_yaxes(title = "Cummulative Return (in %)")
fig.update_layout(width = 950, height = 700, title_x = 0.5, template = "ggplot2")

fig.write_image("Images/Cum_Return_Over_Time.png")

![title](Images/Cum_Return_Over_Time.png)

This plot helps a little more. From the first plot we would expect GOOG to be performing the best. However, this plot clearly shows that TSLA is blowing GOOG out of the water. As is GME, which seemed to be underperforming significantly when compared to GOOG. Let's take a closer look at the historical data of TSLA.

## TSLA Risk Analysis

In [11]:
tsla_asset = portfolio['TSLA']
tsla_return = tsla_asset.pct_change().dropna() * 100

In [12]:
fig = px.line(tsla_return)
fig.update_layout(legend_title = "Asset", title = "Returns (in %) per Day of TSLA")
fig.update_xaxes(title = "Date")
fig.update_yaxes(title = "Return (in %)")
fig.update_layout(width = 950, height = 700, title_x = 0.5, template = "ggplot2")

fig.write_image("Images/Returns_Per_Day_TSLA.png")

![title](Images/Returns_Per_Day_TSLA.png)

From the return per day we can see that the beginning of 2020 was very eventful. This can be explained by the COVID-19 pandemic. Then, the recent visible increase in volatility can also explained by the fear caused by the COVID-19 omicron variant. Let's calculate volatility.

In [13]:
import math

tsla_daily_volatility = tsla_return.std()
print("The average daily volatility is", tsla_daily_volatility)

tsla_monthly_volatility = math.sqrt(21) * tsla_daily_volatility
print("The average monthly volatility is", tsla_monthly_volatility)

tsla_annual_volatility = math.sqrt(252) * tsla_daily_volatility
print("The average annual volatility is", tsla_annual_volatility)

The average daily volatility is 3.625287137572878
The average monthly volatility is 16.613152723877498
The average annual volatility is 57.54964918331424


From here it seems useful to take a look at the distribution of the returns as well as the VaR and cVaR.

In [14]:
fig = px.histogram(tsla_return)
fig.update_layout(legend_title = "", title = "Returns (in %) of TSLA")
fig.update_xaxes(title = "Returns (in %)")
fig.update_yaxes(title = "Frequency of Returns")
fig.update_layout(width = 950, height = 700, title_x = 0.5, template = "ggplot2")

fig.write_image("Images/Returns_TSLA_Hist.png")

![title](Images/Returns_TSLA_Hist.png)

As can be seen, the returns are relatively normal distributed (possibly t-distribution). For now, we are going to assume a normal distribution:

In [15]:
## Calculate VaR(90) and CVaR(90)
var_90 = np.percentile(tsla_return, 10)
cvar_90 = tsla_return[tsla_return <= var_90].mean()

## Calculate VaR(95) and CVaR(95)
var_95 = np.percentile(tsla_return, 5)
cvar_95 = tsla_return[tsla_return <= var_95].mean()

## Calculate VaR(99) and CVaR(99)
var_99 = np.percentile(tsla_return, 1)
cvar_99 = tsla_return[tsla_return <= var_99].mean()


print("VaR(90):",var_90)
print("CVaR(90):", cvar_90)

print("")

print("VaR(90):",var_95)
print("CVaR(90):", cvar_95)

print("")

print("VaR(99):",var_99)
print("CVaR(99):", cvar_99)

VaR(90): -3.354413502899729
CVaR(90): -5.983841024719736

VaR(90): -4.977261277312514
CVaR(90): -7.902930921500541

VaR(99): -8.974423869669101
CVaR(99): -13.277236397085636


Let's add these values to the histogram!

In [16]:
fig.update_yaxes(range = [0,150])

fig.add_trace(go.Scatter(x=(var_90,var_90), y = (0,200), mode = 'lines', line_color='Red', name = "VaR(90)"))
fig.add_trace(go.Scatter(x=(cvar_90,cvar_90),y = (0,200), line_dash='dash', line_color='Red',name = "CVaR(90)"))


fig.add_trace(go.Scatter(x=(var_95,var_95), y = (0,200), mode = 'lines', line_color='Green', name = "VaR(95)"))
fig.add_trace(go.Scatter(x=(cvar_95,cvar_95),y = (0,200), line_dash='dash', line_color='Green',name = "CVaR(95)"))

fig.add_trace(go.Scatter(x=(var_99,var_99), y = (0,200), mode = 'lines', line_color='Purple', name = "VaR(99)"))
fig.add_trace(go.Scatter(x=(cvar_99,cvar_99),y = (0,200), line_dash='dash', line_color='Purple',name = "CVaR(99)"))

fig.write_image("Images/Returns_TSLA_Hist_VaR.png")

![title](Images/Returns_TSLA_Hist_VaR.png)

Great, this plot is giving us a lot of information. Let's focus only on VaR(90). The value is telling us that (historically) there is a probability of 90 percent that the losses don't exceed -3.35%. If they do, there is a 90% probabilty the losses don't exceed 5.98%. This is an example of very simple and straight forward risk management. But before we dive into more detail only focusing on the TSLA assest, let's look at our whole portfolio.

## Hedging

Hedging is mostly used to offset any losses. At first, we will look at how the asset price has an influence on the cost of a put option. To do so we will use the Black Scholes method of pricing options.

In [17]:
## Defining Black Scholes functions 

# The Black-Scholes option pricing formula for European options
# Adapted from: Black, F., & Scholes, M. (1973). The Pricing of Options and Corporate Liabilities. Journal of Political Economy, 81(3), 637-654, www.jstor.org/stable/1831029.
# MIT License
# Copyright (c) 2019 Jamsheed Shorish

import numpy as np
from scipy.stats import norm


# Auxiliary function for d_one risk-adjusted probability
def d11(S, X, T, r, sigma):
    """This is an auxiliary function and is not intended to be called externally."""
    return (np.log(S/X) + (r + 0.5 * sigma**2)*T) / (sigma * np.sqrt(T))

# Auxiliary function for d_two risk-adjusted probability    
def d21(d1, T, sigma):
    """This is an auxiliary function and is not intended to be called externally."""
    return d1 - sigma * np.sqrt(T)

# Definition of the Black-Scholes delta function    
def bs_delta(S, X, T, r, sigma, option_type):
    """Compute the delta of the Black-Scholes option pricing formula.
    
    Arguments:
    S           -- the current spot price of the underlying stock
    X           -- the option strike price
    T           -- the time until maturity (in fractions of a year)
    r           -- the risk-free interest rate 
    sigma       -- the returns volatility of the underlying stock
    option_type -- the option type, either 'call' or 'put'
    
    Returns: a numpy.float_ representing the delta value
    Exceptions raised: ValueError if option_type is not 'call' or 'put'
    """
    if option_type == 'call':
        return norm.cdf(d11(S, X, T, r, sigma))
    elif option_type == 'put':
        return norm.cdf(-d11(S, X, T, r, sigma))
    else:
        # Raise an error if the option_type is neither a call nor a put
        raise ValueError("Option type is either 'call' or 'put'.")

# Definition of the Black-Scholes European option pricing formula    
def black_scholes(S, X, T, r, sigma, option_type):
    """Price a European option using the Black-Scholes option pricing formula.
    
    Arguments:
    S           -- the current spot price of the underlying stock
    X           -- the option strike price
    T           -- the time until maturity (in fractions of a year)
    r           -- the risk-free interest rate 
    sigma       -- the returns volatility of the underlying stock
    option_type -- the option type, either 'call' or 'put'
    
    Returns: a numpy.float_ representing the option value
    Exceptions raised: ValueError if option_type is not 'call' or 'put'
    """
    d_one = d11(S, X, T, r, sigma)
    d_two = d21(d_one, T, sigma)
    if option_type == 'call':
        return S * norm.cdf(d_one) - np.exp(-r * T) * X * norm.cdf(d_two)
    elif option_type == 'put':
        return -(S * norm.cdf(-d_one) - np.exp(-r * T) * X * norm.cdf(-d_two))
    else:
        # Raise an error if the option_type is neither a call nor a put
        raise ValueError("Option type is either 'call' or 'put'.")

## Defining Black Scholes functions

In [18]:
tsla_price_last_year = tsla_asset.tail(252)
option_values_1500 = np.zeros(tsla_price_last_year.size)

## Strike price is choosen arbitrarily

for i,S in enumerate(tsla_price_last_year.values):
    option_values_1500[i] = black_scholes(S = S, X = 1500, T = 1, r = 0.02, 
                        sigma = tsla_annual_volatility/100, option_type = 'put')

In [19]:
fig = go.Figure()

fig.add_trace(go.Scatter(x=np.array(range(252)), y = option_values_1500, mode = 'lines',
                         name = "Put Option:<br>Strike = $1500"))

fig.add_trace(go.Scatter(x=np.array(range(252)), y = tsla_price_last_year, mode = 'lines', name = "TSLA"))
fig.update_layout(legend_title = "Asset", title = "Correlation between Asset Price and Put Value")
fig.update_xaxes(title = "Time")
fig.update_yaxes(title = "Value (in $)")
fig.update_layout(width = 950, height = 700, title_x = 0.5, template = "ggplot2")

fig.write_image("Images/TSLA_Put_New.png")

![title](Images/TSLA_Put_New.png)

Here we can clearly see the connection between the assets price and the put price. If the asset prices rises, the value of the put option lowers and vice versa. This is exactly what we need for our hedging plan.

Now we can calculate our option delta. This will allow us to show that with each change in the asset/option price we offset our loss. Let's assume our assets value falls from 1100 to 1095.

In [20]:
value = black_scholes(S = 1100, X = 1500, T = 1, r = 0.02, 
                      sigma = tsla_annual_volatility/100, option_type = "put")

delta = bs_delta(S = 1100, X = 1500, T = 1, r = 0.02, 
                 sigma = tsla_annual_volatility/100, option_type = "put")

value_change = black_scholes(S = 1095, X = 1500, T = 1, r = 0.02, 
                             sigma = tsla_annual_volatility/100, option_type = "put") - value

print( (1095 - 1100) + (1/delta) * value_change )

0.013151183222512408


Generally the aim is for delta neutrality = 0 - this would mean we offset any drop in the assets price with the same increase in the put value. We can see that this is the case in our example for TSLA. Therefore, this put option would be very useful to hedge our TSLA stock.

## Best Suited Portfolio Weights - GMV and MSR and The Efficient Frontier

Let's determine the portfolio weights for the best valued portfolio and the portfolio with the least risk - we can use the efficient frontier to do so! We are going to include all assets in the efficient frontier.

In [21]:
annualized_asset_returns = (1+asset_returns.mean())**252 - 1
annualized_cov = (asset_returns).cov()*252
risk_free_rate=0
n_portfolios = 2000

plot_pairs = []
weights_list = []

np.random.seed(2)

for i in range(n_portfolios):
    weights = np.random.rand(5)
    weights = weights/sum(weights)
    weights_list.append([weights])
    portfolio_E_Variance = np.sqrt(np.dot(weights.T, np.dot(annualized_cov, weights)))
    portfolio_E_Return = annualized_asset_returns.multiply(weights, axis = 0).sum()
    plot_pairs.append([portfolio_E_Return, portfolio_E_Variance])
    
    
plot_pairs = np.array(plot_pairs)
sharpe_ratio = np.array((plot_pairs[:,0] - risk_free_rate)/plot_pairs[:,1])
fig = go.Figure()
fig.add_trace(go.Scatter(x=plot_pairs[:,1], y=plot_pairs[:,0], 
                         marker = dict(color = sharpe_ratio,showscale=True,
                                       colorbar=dict(title="Sharpe<br>Ratio")),mode='markers'))
fig.update_layout(title = "The Efficient Frontier",title_x = 0.5)
fig.update_yaxes(title = "Annualized Return")
fig.update_xaxes(title = "Annualized Risk (Volatility)")
fig.write_image("Images/The_Efficient_Frontier_New.png")

![title](Images/The_Efficient_Frontier_New.png)

From here we can see the efficient frontier. We can pick a point on the efficient frontier following the basic rule: The higher the return, the higher the risk. The lower the return, the lower the risk. Let's calculate the MSR and GMV.

In [22]:
indx_msr = sharpe_ratio.argsort()[len(sharpe_ratio)-1]

print("The maximum sharpe ratio is:", sharpe_ratio[indx_msr])

print("The associated weights are:",weights_list[indx_msr])

The maximum sharpe ratio is: 2.1940235623317452
The associated weights are: [array([0.39209185, 0.14340418, 0.00617222, 0.445952  , 0.01237976])]


In [23]:
indx_gmv = plot_pairs[:,1].argsort()[0]

print("The minimum volatility is:",plot_pairs[indx_gmv][1])

print("The associated weights are:",weights_list[indx_gmv])

The minimum volatility is: 0.2502319371101916
The associated weights are: [array([0.02928756, 0.04248366, 0.01379509, 0.4640012 , 0.45043249])]


In [24]:
print(tickers)

['TSLA', 'GME', 'AMC', 'MSFT', 'GOOG']


This is giving us some great insight. If we want a great performing portfolio (high sharpe ratio) we would wanna hold mostly MSFT and TSLA with some GME but almost no GOOG and GME. However, this sharpe ratio is historical. This means, just because the historical sharpe ratio is good - the future is not guaranteed to be good. When looking at the GMV, we would mostly hold MSTF and GOOG. This makes sense since the historical growth has been really steady for both stocks - indicating low risk - but also low reward. 

Let's asses our VaR and CVaR again! 

In [25]:
gmv_weights = [0.02928756, 0.04248366, 0.01379509, 0.4640012 , 0.45043249]

gmv_portfolio = (portfolio * gmv_weights).sum(axis=1)
gmv_portfolio_pct_returns = gmv_portfolio.pct_change().dropna() * 100


In [26]:
fig = px.line(gmv_portfolio_pct_returns)
fig.update_layout(legend_title = "Asset", title = "Returns (in %) per Day of our GMV")
fig.update_xaxes(title = "Date")
fig.update_yaxes(title = "Return (in %)")
fig.update_layout(width = 950, height = 700, title_x = 0.5, template = "ggplot2",showlegend=False)

fig.write_image("Images/Returns_Per_Day_GMV_New.png")

![title](Images/Returns_Per_Day_GMV_New.png)

We have choosen the GMV this means that the volatility overall is very low. However, there is a high volatility peak around the beginning of 2020 which is unavoidable due to COVID-19.

In [27]:
fig = px.histogram(gmv_portfolio_pct_returns)
fig.update_layout(legend_title = "", title = "Returns (in %) of GMV")
fig.update_xaxes(title = "Returns (in %)")
fig.update_yaxes(title = "Frequency of Returns")
fig.update_layout(width = 950, height = 700, title_x = 0.5, template = "ggplot2", showlegend = False)

fig.write_image("Images/Returns_GMV_Hist_New.png")

![title](Images/Returns_GMV_Hist_New.png)

Let's add our *historical* VaR and CVaR.

In [28]:
## Calculate VaR(90) and CVaR(90)
var_90 = np.percentile(gmv_portfolio_pct_returns, 10)
cvar_90 = tsla_return[gmv_portfolio_pct_returns <= var_90].mean()

## Calculate VaR(95) and CVaR(95)
var_95 = np.percentile(gmv_portfolio_pct_returns, 5)
cvar_95 = tsla_return[gmv_portfolio_pct_returns <= var_95].mean()

## Calculate VaR(99) and CVaR(99)
var_99 = np.percentile(gmv_portfolio_pct_returns, 1)
cvar_99 = tsla_return[gmv_portfolio_pct_returns <= var_99].mean()


print("VaR(90):",var_90)
print("CVaR(90):", cvar_90)

print("")

print("VaR(90):",var_95)
print("CVaR(90):", cvar_95)

print("")

print("VaR(99):",var_99)
print("CVaR(99):", cvar_99)

fig.update_yaxes(range = [0,150])

fig.add_trace(go.Scatter(x=(var_90,var_90), y = (0,200), mode = 'lines', line_color='Red', name = "VaR(90)"))
fig.add_trace(go.Scatter(x=(cvar_90,cvar_90),y = (0,200), line_dash='dash', line_color='Red',name = "CVaR(90)"))


fig.add_trace(go.Scatter(x=(var_95,var_95), y = (0,200), mode = 'lines', line_color='Green', name = "VaR(95)"))
fig.add_trace(go.Scatter(x=(cvar_95,cvar_95),y = (0,200), line_dash='dash', line_color='Green',name = "CVaR(95)"))

fig.add_trace(go.Scatter(x=(var_99,var_99), y = (0,200), mode = 'lines', line_color='Purple', name = "VaR(99)"))
fig.add_trace(go.Scatter(x=(cvar_99,cvar_99),y = (0,200), line_dash='dash', line_color='Purple',name = "CVaR(99)"))

fig.write_image("Images/Returns_GMV_Hist_VaR.png")

VaR(90): -1.6208245377309805
CVaR(90): -2.9433353757639957

VaR(90): -2.4769765312070344
CVaR(90): -3.726762755595857

VaR(99): -4.7602899224502755
CVaR(99): -5.731114971323832


![title](Images/Returns_GMV_Hist_VaR.png)

However, this does not help us a lot. We know historically, it is expected to not loose more than 1.62% in 95% of cases. But this is the past - we need the future. We can use a Monte Carlo Estimation to generate possible futures and use those to calculate a better VaR and CVaR. First we need the daily portfolio variance.

In [29]:
portfolio_returns = portfolio.pct_change()
portfolio_cov = portfolio_returns.cov()
portfolio_vol_daily = np.sqrt(np.dot(np.dot(np.array(gmv_weights), portfolio_cov), np.array(gmv_weights).T))
portfolio_avg_ret = np.mean(gmv_portfolio_pct_returns)

In [30]:
print("The average daily return is:",portfolio_avg_ret, "%")
print("The average daily volatility is:",portfolio_vol_daily*100,"%")

The average daily return is: 0.10411631037108311 %
The average daily volatility is: 1.5763130415588282 %


If we compare that to the daily volatility of the TSLA stock we see a large difference (1.57% (GMV) vs. 3.62% (TSLA)). If assuming that the distribution of the returns in the future is normal, we can choose mu to be our average daily portfolio return and the standard deviation to be the daily volatility. We are going to predict 252 days (one trading year). Then we can simulate the returns:

In [31]:
mu = portfolio_avg_ret
sd = portfolio_vol_daily*100
T = 252

sim_returns = []

## Each simulation is one year
for i in range(100):

    rand_rets = np.random.normal(mu, sd, T)
    
    sim_returns = np.concatenate([sim_returns, rand_rets])

We can use the simulated returns to calculate the most probable VaR(95) and CVaR(95).

In [32]:
## Calculate VaR(90) and CVaR(90)
var_95 = np.percentile(sim_returns, 5)
cvar_95 = sim_returns[sim_returns <= var_95].mean()

print("VaR(95):",var_95)
print("CVaR(95):", cvar_95)

VaR(95): -2.475763717610942
CVaR(95): -3.1388585560309212


We can see that using the Monte Carlo estimation we get similar VaR(95) and CVaR(95) values for the *upcoming* year -  which makes sense.

## The GARCH Model for Prediction and VaR/CVaR

## Predicting Asset Prices using Neural Networks and Deep Learning

In this section we will use 4 of our stock prices to predict the price of the last/5th one using Deep Learning and Tensorflow. 

For our training set we will keep all prices except the last - most recent - 6 Month. This month can be used for testing. Let's predict the price of GOOG.

In [68]:
train_set = portfolio.drop(portfolio.tail(21*6).index)
test_set = portfolio.tail(21*6)

X_train = train_set[['TSLA','GME','AMC','MSFT']]
y_train = train_set[['GOOG']]

X_test = test_set[['TSLA','GME','AMC','MSFT']]
y_test = test_set[['GOOG']]

After creating the train and test set we can prepare the model and train it.

In [91]:
import tensorflow as tf
from tensorflow import keras
from tensorflow.keras import layers

model = keras.Sequential()
model.add(layers.Dense(500, input_dim=4, activation='sigmoid'))
model.add(layers.Dense(100, activation='relu'))
model.add(layers.Dense(100, activation='relu'))
model.add(layers.Dense(100, activation='relu'))
model.add(layers.Dense(50, activation='relu'))
model.add(layers.Dense(50, activation='relu'))
model.add(layers.Dense(50, activation='relu'))
model.add(layers.Dense(25, activation='relu'))
model.add(layers.Dense(1))

model.compile(loss='mean_squared_logarithmic_error', optimizer='rmsprop')
model.fit(X_train, y_train, epochs=6000)

Epoch 1/6000


2022-01-23 17:27:44.983919: I tensorflow/core/grappler/optimizers/custom_graph_optimizer_registry.cc:112] Plugin optimizer for device_type GPU is enabled.


44/44 [==============================] - 2s 12ms/step - loss: 9.0766
Epoch 2/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.1029
Epoch 3/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0296
Epoch 4/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0168
Epoch 5/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0133
Epoch 6/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0138
Epoch 7/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0114
Epoch 8/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0116
Epoch 9/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0109
Epoch 10/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0092
Epoch 11/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0103
Epoch 12/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0102
Epoch 13/6000
44/44 [================

44/44 [==============================] - 0s 9ms/step - loss: 0.0047
Epoch 101/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0055
Epoch 102/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0049
Epoch 103/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0051
Epoch 104/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0050
Epoch 105/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0044
Epoch 106/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0050
Epoch 107/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0050
Epoch 108/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0052
Epoch 109/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0049
Epoch 110/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0052
Epoch 111/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0048
Epoch 112/6000
44/4

44/44 [==============================] - 0s 9ms/step - loss: 0.0038
Epoch 199/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0038
Epoch 200/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0035
Epoch 201/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0038
Epoch 202/6000
44/44 [==============================] - 0s 10ms/step - loss: 0.0033
Epoch 203/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0036
Epoch 204/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0036
Epoch 205/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0037
Epoch 206/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0034
Epoch 207/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0032
Epoch 208/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0040
Epoch 209/6000
44/44 [==============================] - 0s 10ms/step - loss: 0.0032
Epoch 210/6000
44

44/44 [==============================] - 0s 9ms/step - loss: 0.0031
Epoch 297/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0027
Epoch 298/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0033
Epoch 299/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0027
Epoch 300/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0029
Epoch 301/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0032
Epoch 302/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0027
Epoch 303/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0031
Epoch 304/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0029
Epoch 305/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0028
Epoch 306/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0029
Epoch 307/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0032
Epoch 308/6000
44/4

44/44 [==============================] - 0s 9ms/step - loss: 0.0026
Epoch 395/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0025
Epoch 396/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0026
Epoch 397/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0023
Epoch 398/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0028
Epoch 399/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0027
Epoch 400/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0023
Epoch 401/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0027
Epoch 402/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0025
Epoch 403/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0026
Epoch 404/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0026
Epoch 405/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0026
Epoch 406/6000
44/4

44/44 [==============================] - 0s 10ms/step - loss: 0.0025
Epoch 493/6000
44/44 [==============================] - 0s 10ms/step - loss: 0.0025
Epoch 494/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0024
Epoch 495/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0022
Epoch 496/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0025
Epoch 497/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0024
Epoch 498/6000
44/44 [==============================] - 0s 10ms/step - loss: 0.0023
Epoch 499/6000
44/44 [==============================] - 0s 10ms/step - loss: 0.0024
Epoch 500/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0023
Epoch 501/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0023
Epoch 502/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0022
Epoch 503/6000
44/44 [==============================] - 0s 10ms/step - loss: 0.0023
Epoch 504/6000

44/44 [==============================] - 0s 9ms/step - loss: 0.0023
Epoch 591/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0021
Epoch 592/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0022
Epoch 593/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0023
Epoch 594/6000
44/44 [==============================] - 0s 10ms/step - loss: 0.0022
Epoch 595/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0022
Epoch 596/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0022
Epoch 597/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0022
Epoch 598/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0022
Epoch 599/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0023
Epoch 600/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0022
Epoch 601/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0023
Epoch 602/6000
44/

44/44 [==============================] - 0s 9ms/step - loss: 0.0021
Epoch 689/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0022
Epoch 690/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0019
Epoch 691/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0020
Epoch 692/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0021
Epoch 693/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0019
Epoch 694/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0021
Epoch 695/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0021
Epoch 696/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0020
Epoch 697/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0023
Epoch 698/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0021
Epoch 699/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0022
Epoch 700/6000
44/4

44/44 [==============================] - 0s 9ms/step - loss: 0.0021
Epoch 787/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0020
Epoch 788/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0020
Epoch 789/6000
44/44 [==============================] - 0s 10ms/step - loss: 0.0018
Epoch 790/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0020
Epoch 791/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0020
Epoch 792/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0020
Epoch 793/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0020
Epoch 794/6000
44/44 [==============================] - 0s 10ms/step - loss: 0.0020
Epoch 795/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0020
Epoch 796/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0021
Epoch 797/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0019
Epoch 798/6000
44

44/44 [==============================] - 0s 9ms/step - loss: 0.0018
Epoch 885/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0021
Epoch 886/6000
44/44 [==============================] - 1s 11ms/step - loss: 0.0019
Epoch 887/6000
44/44 [==============================] - 1s 13ms/step - loss: 0.0019
Epoch 888/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0018
Epoch 889/6000
44/44 [==============================] - 0s 10ms/step - loss: 0.0020
Epoch 890/6000
44/44 [==============================] - 0s 10ms/step - loss: 0.0019
Epoch 891/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0019
Epoch 892/6000
44/44 [==============================] - 0s 10ms/step - loss: 0.0018
Epoch 893/6000
44/44 [==============================] - 0s 10ms/step - loss: 0.0019
Epoch 894/6000
44/44 [==============================] - 0s 10ms/step - loss: 0.0018
Epoch 895/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0020
Epoch 896/60

44/44 [==============================] - 0s 9ms/step - loss: 0.0019
Epoch 983/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0019
Epoch 984/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0019
Epoch 985/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0017
Epoch 986/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0020
Epoch 987/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0016
Epoch 988/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0019
Epoch 989/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0018
Epoch 990/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0019
Epoch 991/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0019
Epoch 992/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0018
Epoch 993/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0018
Epoch 994/6000
44/4

44/44 [==============================] - 0s 9ms/step - loss: 0.0019
Epoch 1080/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0019
Epoch 1081/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0018
Epoch 1082/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0017
Epoch 1083/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0019
Epoch 1084/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0017
Epoch 1085/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0018
Epoch 1086/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0018
Epoch 1087/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0017
Epoch 1088/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0019
Epoch 1089/6000
44/44 [==============================] - 0s 10ms/step - loss: 0.0018
Epoch 1090/6000
44/44 [==============================] - 0s 10ms/step - loss: 0.0018
Epoch 

44/44 [==============================] - 0s 9ms/step - loss: 0.0017
Epoch 1177/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0018
Epoch 1178/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0016
Epoch 1179/6000
44/44 [==============================] - 0s 10ms/step - loss: 0.0018
Epoch 1180/6000
44/44 [==============================] - 0s 10ms/step - loss: 0.0017
Epoch 1181/6000
44/44 [==============================] - 1s 13ms/step - loss: 0.0018
Epoch 1182/6000
44/44 [==============================] - 1s 12ms/step - loss: 0.0016
Epoch 1183/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0018
Epoch 1184/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0016
Epoch 1185/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0018
Epoch 1186/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0017
Epoch 1187/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0017
Epoc

44/44 [==============================] - 0s 9ms/step - loss: 0.0017
Epoch 1274/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0015
Epoch 1275/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0018
Epoch 1276/6000
44/44 [==============================] - 0s 10ms/step - loss: 0.0016
Epoch 1277/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0016
Epoch 1278/6000
44/44 [==============================] - 0s 10ms/step - loss: 0.0016
Epoch 1279/6000
44/44 [==============================] - 0s 10ms/step - loss: 0.0017
Epoch 1280/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0018
Epoch 1281/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0016
Epoch 1282/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0017
Epoch 1283/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0016
Epoch 1284/6000
44/44 [==============================] - 0s 10ms/step - loss: 0.0016
Epoc

44/44 [==============================] - 0s 9ms/step - loss: 0.0017
Epoch 1371/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0015
Epoch 1372/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0015
Epoch 1373/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0016
Epoch 1374/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0015
Epoch 1375/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0016
Epoch 1376/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0017
Epoch 1377/6000
44/44 [==============================] - 0s 10ms/step - loss: 0.0016
Epoch 1378/6000
44/44 [==============================] - 0s 10ms/step - loss: 0.0016
Epoch 1379/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0016
Epoch 1380/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0017
Epoch 1381/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0015
Epoch 

44/44 [==============================] - 0s 9ms/step - loss: 0.0016
Epoch 1468/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0015
Epoch 1469/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0016
Epoch 1470/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0016
Epoch 1471/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0015
Epoch 1472/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0016
Epoch 1473/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0015
Epoch 1474/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0016
Epoch 1475/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0017
Epoch 1476/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0014
Epoch 1477/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0015
Epoch 1478/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0016
Epoch 14

44/44 [==============================] - 0s 10ms/step - loss: 0.0015
Epoch 1565/6000
44/44 [==============================] - 0s 10ms/step - loss: 0.0016
Epoch 1566/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0015
Epoch 1567/6000
44/44 [==============================] - 0s 10ms/step - loss: 0.0015
Epoch 1568/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0016
Epoch 1569/6000
44/44 [==============================] - 0s 11ms/step - loss: 0.0015
Epoch 1570/6000
44/44 [==============================] - 1s 14ms/step - loss: 0.0015
Epoch 1571/6000
44/44 [==============================] - 1s 13ms/step - loss: 0.0015
Epoch 1572/6000
44/44 [==============================] - 0s 10ms/step - loss: 0.0015
Epoch 1573/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0014
Epoch 1574/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0015
Epoch 1575/6000
44/44 [==============================] - 0s 10ms/step - loss: 0.0015


44/44 [==============================] - 0s 9ms/step - loss: 0.0015
Epoch 1662/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0015
Epoch 1663/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0014
Epoch 1664/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0014
Epoch 1665/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0015
Epoch 1666/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0014
Epoch 1667/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0013
Epoch 1668/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0015
Epoch 1669/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0014
Epoch 1670/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0015
Epoch 1671/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0014
Epoch 1672/6000
44/44 [==============================] - 0s 10ms/step - loss: 0.0014
Epoch 1

44/44 [==============================] - 0s 9ms/step - loss: 0.0015
Epoch 1759/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0014
Epoch 1760/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0015
Epoch 1761/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0014
Epoch 1762/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0014
Epoch 1763/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0014
Epoch 1764/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0014
Epoch 1765/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0015
Epoch 1766/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0015
Epoch 1767/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0014
Epoch 1768/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0014
Epoch 1769/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0016
Epoch 17

44/44 [==============================] - 0s 9ms/step - loss: 0.0014
Epoch 1856/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0015
Epoch 1857/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0014
Epoch 1858/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0014
Epoch 1859/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0013
Epoch 1860/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0015
Epoch 1861/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0014
Epoch 1862/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0014
Epoch 1863/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0015
Epoch 1864/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0013
Epoch 1865/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0014
Epoch 1866/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0014
Epoch 18

44/44 [==============================] - 0s 9ms/step - loss: 0.0013
Epoch 1953/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0014
Epoch 1954/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0013
Epoch 1955/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0014
Epoch 1956/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0015
Epoch 1957/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0013
Epoch 1958/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0013
Epoch 1959/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0015
Epoch 1960/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0014
Epoch 1961/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0014
Epoch 1962/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0013
Epoch 1963/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0014
Epoch 19

44/44 [==============================] - 0s 9ms/step - loss: 0.0013
Epoch 2050/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0013
Epoch 2051/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0013
Epoch 2052/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0012
Epoch 2053/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0014
Epoch 2054/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0015
Epoch 2055/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0012
Epoch 2056/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0013
Epoch 2057/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0014
Epoch 2058/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0014
Epoch 2059/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0013
Epoch 2060/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0013
Epoch 20

44/44 [==============================] - 0s 9ms/step - loss: 0.0013
Epoch 2147/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0013
Epoch 2148/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0013
Epoch 2149/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0014
Epoch 2150/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0013
Epoch 2151/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0012
Epoch 2152/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0014
Epoch 2153/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0012
Epoch 2154/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0013
Epoch 2155/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0013
Epoch 2156/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0013
Epoch 2157/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0012
Epoch 21

44/44 [==============================] - 0s 9ms/step - loss: 0.0013
Epoch 2244/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0014
Epoch 2245/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0013
Epoch 2246/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0012
Epoch 2247/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0013
Epoch 2248/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0012
Epoch 2249/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0013
Epoch 2250/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0012
Epoch 2251/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0012
Epoch 2252/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0013
Epoch 2253/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0013
Epoch 2254/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0013
Epoch 22

44/44 [==============================] - 0s 8ms/step - loss: 0.0013
Epoch 2341/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0012
Epoch 2342/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0012
Epoch 2343/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0011
Epoch 2344/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0011
Epoch 2345/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0012
Epoch 2346/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0012
Epoch 2347/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0013
Epoch 2348/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0011
Epoch 2349/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0013
Epoch 2350/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0012
Epoch 2351/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0012
Epoch 23

44/44 [==============================] - 0s 9ms/step - loss: 0.0011
Epoch 2438/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0011
Epoch 2439/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0012
Epoch 2440/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0011
Epoch 2441/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0011
Epoch 2442/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0011
Epoch 2443/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0013
Epoch 2444/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0010
Epoch 2445/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0013
Epoch 2446/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0011
Epoch 2447/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0012
Epoch 2448/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0011
Epoch 24

44/44 [==============================] - 0s 9ms/step - loss: 0.0012
Epoch 2535/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0011
Epoch 2536/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0011
Epoch 2537/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0011
Epoch 2538/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0011
Epoch 2539/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0011
Epoch 2540/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0011
Epoch 2541/6000
44/44 [==============================] - 0s 8ms/step - loss: 0.0012
Epoch 2542/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0010
Epoch 2543/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0011
Epoch 2544/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0011
Epoch 2545/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0011
Epoch 25

44/44 [==============================] - 0s 9ms/step - loss: 0.0012
Epoch 2632/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0012
Epoch 2633/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0011
Epoch 2634/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0011
Epoch 2635/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0011
Epoch 2636/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0011
Epoch 2637/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0012
Epoch 2638/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.8663e-04
Epoch 2639/6000
44/44 [==============================] - 1s 15ms/step - loss: 0.0012
Epoch 2640/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0010
Epoch 2641/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0011
Epoch 2642/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0010
Epo

44/44 [==============================] - 0s 9ms/step - loss: 0.0011
Epoch 2729/6000
44/44 [==============================] - 0s 8ms/step - loss: 9.9947e-04
Epoch 2730/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0011
Epoch 2731/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.6691e-04
Epoch 2732/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0011
Epoch 2733/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0011
Epoch 2734/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.9898e-04
Epoch 2735/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0011
Epoch 2736/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0012
Epoch 2737/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0010
Epoch 2738/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.8643e-04
Epoch 2739/6000
44/44 [==============================] - 0s 9ms/step - loss:

44/44 [==============================] - 0s 9ms/step - loss: 9.8697e-04
Epoch 2825/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.7796e-04
Epoch 2826/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0011
Epoch 2827/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0010
Epoch 2828/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0011
Epoch 2829/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0010
Epoch 2830/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0011
Epoch 2831/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0012
Epoch 2832/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.4080e-04
Epoch 2833/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0011
Epoch 2834/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0010
Epoch 2835/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0

44/44 [==============================] - 0s 9ms/step - loss: 0.0010
Epoch 2921/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0011
Epoch 2922/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.6758e-04
Epoch 2923/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0010
Epoch 2924/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0010
Epoch 2925/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.5825e-04
Epoch 2926/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.7628e-04
Epoch 2927/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0011
Epoch 2928/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.5897e-04
Epoch 2929/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0011
Epoch 2930/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0011
Epoch 2931/6000
44/44 [==============================] - 0s 9ms/step - loss:

44/44 [==============================] - 0s 9ms/step - loss: 0.0010
Epoch 3016/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.5941e-04
Epoch 3017/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0010
Epoch 3018/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0011
Epoch 3019/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.2460e-04
Epoch 3020/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0010
Epoch 3021/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.0420e-04
Epoch 3022/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0010
Epoch 3023/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.3576e-04
Epoch 3024/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.6949e-04
Epoch 3025/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0010
Epoch 3026/6000
44/44 [==============================] - 0s 9ms/step - l

44/44 [==============================] - 0s 9ms/step - loss: 9.7550e-04
Epoch 3111/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.3548e-04
Epoch 3112/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.6332e-04
Epoch 3113/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.8990e-04
Epoch 3114/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0010
Epoch 3115/6000
44/44 [==============================] - 0s 10ms/step - loss: 9.5631e-04
Epoch 3116/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.5421e-04
Epoch 3117/6000
44/44 [==============================] - 0s 10ms/step - loss: 9.8648e-04
Epoch 3118/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.7521e-04
Epoch 3119/6000
44/44 [==============================] - 0s 10ms/step - loss: 9.6277e-04
Epoch 3120/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.3839e-04
Epoch 3121/6000
44/44 [==========================

44/44 [==============================] - 0s 8ms/step - loss: 9.9440e-04
Epoch 3205/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.0974e-04
Epoch 3206/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.7018e-04
Epoch 3207/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.5076e-04
Epoch 3208/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.3415e-04
Epoch 3209/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0010
Epoch 3210/6000
44/44 [==============================] - 0s 8ms/step - loss: 9.3415e-04
Epoch 3211/6000
44/44 [==============================] - 0s 8ms/step - loss: 8.8925e-04
Epoch 3212/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0010
Epoch 3213/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.7297e-04
Epoch 3214/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.3654e-04
Epoch 3215/6000
44/44 [==============================] -

44/44 [==============================] - 0s 9ms/step - loss: 9.9141e-04
Epoch 3298/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.6283e-04
Epoch 3299/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0010
Epoch 3300/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0010
Epoch 3301/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.0858e-04
Epoch 3302/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.2873e-04
Epoch 3303/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0010
Epoch 3304/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.8511e-04
Epoch 3305/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.0452e-04
Epoch 3306/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0010
Epoch 3307/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.2712e-04
Epoch 3308/6000
44/44 [==============================] - 0s 9ms/

44/44 [==============================] - 0s 9ms/step - loss: 8.8751e-04
Epoch 3392/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.9286e-04
Epoch 3393/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.5107e-04
Epoch 3394/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.4260e-04
Epoch 3395/6000
44/44 [==============================] - 0s 8ms/step - loss: 9.3995e-04
Epoch 3396/6000
44/44 [==============================] - 0s 8ms/step - loss: 9.6924e-04
Epoch 3397/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.8615e-04
Epoch 3398/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.0155e-04
Epoch 3399/6000
44/44 [==============================] - 0s 9ms/step - loss: 0.0010
Epoch 3400/6000
44/44 [==============================] - 0s 8ms/step - loss: 8.6345e-04
Epoch 3401/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.5625e-04
Epoch 3402/6000
44/44 [=============================

44/44 [==============================] - 0s 9ms/step - loss: 8.9428e-04
Epoch 3485/6000
44/44 [==============================] - 0s 8ms/step - loss: 8.9371e-04
Epoch 3486/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.1072e-04
Epoch 3487/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.5306e-04
Epoch 3488/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.2687e-04
Epoch 3489/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.4934e-04
Epoch 3490/6000
44/44 [==============================] - 0s 8ms/step - loss: 8.9792e-04
Epoch 3491/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.6590e-04
Epoch 3492/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.1908e-04
Epoch 3493/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.1433e-04
Epoch 3494/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.8252e-04
Epoch 3495/6000
44/44 [=========================

44/44 [==============================] - 0s 9ms/step - loss: 9.1840e-04
Epoch 3578/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.0838e-04
Epoch 3579/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.2406e-04
Epoch 3580/6000
44/44 [==============================] - 0s 10ms/step - loss: 9.8300e-04
Epoch 3581/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.5543e-04
Epoch 3582/6000
44/44 [==============================] - 0s 10ms/step - loss: 8.0505e-04
Epoch 3583/6000
44/44 [==============================] - 0s 10ms/step - loss: 8.8587e-04
Epoch 3584/6000
44/44 [==============================] - 0s 11ms/step - loss: 9.9343e-04
Epoch 3585/6000
44/44 [==============================] - 0s 10ms/step - loss: 8.4777e-04
Epoch 3586/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.3460e-04
Epoch 3587/6000
44/44 [==============================] - 0s 10ms/step - loss: 8.2091e-04
Epoch 3588/6000
44/44 [===================

44/44 [==============================] - 0s 9ms/step - loss: 9.5351e-04
Epoch 3671/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.4017e-04
Epoch 3672/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.1507e-04
Epoch 3673/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.8900e-04
Epoch 3674/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.8813e-04
Epoch 3675/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.2664e-04
Epoch 3676/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.2605e-04
Epoch 3677/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.0309e-04
Epoch 3678/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.2606e-04
Epoch 3679/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.6741e-04
Epoch 3680/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.3909e-04
Epoch 3681/6000
44/44 [=========================

44/44 [==============================] - 0s 9ms/step - loss: 8.8106e-04
Epoch 3763/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.9230e-04
Epoch 3764/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.7651e-04
Epoch 3765/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.3320e-04
Epoch 3766/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.0988e-04
Epoch 3767/6000
44/44 [==============================] - 0s 8ms/step - loss: 9.1398e-04
Epoch 3768/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.1321e-04
Epoch 3769/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.9969e-04
Epoch 3770/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.7865e-04
Epoch 3771/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.8952e-04
Epoch 3772/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.4373e-04
Epoch 3773/6000
44/44 [=========================

44/44 [==============================] - 0s 9ms/step - loss: 8.4458e-04
Epoch 3856/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.1414e-04
Epoch 3857/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.0766e-04
Epoch 3858/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.2129e-04
Epoch 3859/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.4466e-04
Epoch 3860/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.9353e-04
Epoch 3861/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.6557e-04
Epoch 3862/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.5585e-04
Epoch 3863/6000
44/44 [==============================] - 0s 9ms/step - loss: 9.3805e-04
Epoch 3864/6000
44/44 [==============================] - 0s 8ms/step - loss: 8.4866e-04
Epoch 3865/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.1333e-04
Epoch 3866/6000
44/44 [=========================

44/44 [==============================] - 0s 9ms/step - loss: 8.9022e-04
Epoch 3949/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.7191e-04
Epoch 3950/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.6260e-04
Epoch 3951/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.1013e-04
Epoch 3952/6000
44/44 [==============================] - 0s 8ms/step - loss: 8.5146e-04
Epoch 3953/6000
44/44 [==============================] - 0s 8ms/step - loss: 8.6773e-04
Epoch 3954/6000
44/44 [==============================] - 0s 8ms/step - loss: 8.1965e-04
Epoch 3955/6000
44/44 [==============================] - 0s 8ms/step - loss: 7.8856e-04
Epoch 3956/6000
44/44 [==============================] - 0s 8ms/step - loss: 8.6052e-04
Epoch 3957/6000
44/44 [==============================] - 0s 8ms/step - loss: 8.9437e-04
Epoch 3958/6000
44/44 [==============================] - 0s 8ms/step - loss: 8.8416e-04
Epoch 3959/6000
44/44 [=========================

44/44 [==============================] - 0s 9ms/step - loss: 8.5875e-04
Epoch 4042/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.8595e-04
Epoch 4043/6000
44/44 [==============================] - 0s 8ms/step - loss: 8.2784e-04
Epoch 4044/6000
44/44 [==============================] - 0s 8ms/step - loss: 8.4537e-04
Epoch 4045/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.2548e-04
Epoch 4046/6000
44/44 [==============================] - 0s 11ms/step - loss: 8.0887e-04
Epoch 4047/6000
44/44 [==============================] - 0s 11ms/step - loss: 8.6306e-04
Epoch 4048/6000
44/44 [==============================] - 0s 10ms/step - loss: 8.2375e-04
Epoch 4049/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.8123e-04
Epoch 4050/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.7390e-04
Epoch 4051/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.4410e-04
Epoch 4052/6000
44/44 [======================

44/44 [==============================] - 0s 9ms/step - loss: 8.5245e-04
Epoch 4134/6000
44/44 [==============================] - 0s 10ms/step - loss: 7.6527e-04
Epoch 4135/6000
44/44 [==============================] - 0s 10ms/step - loss: 8.9635e-04
Epoch 4136/6000
44/44 [==============================] - 0s 10ms/step - loss: 8.1235e-04
Epoch 4137/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.0541e-04
Epoch 4138/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.8469e-04
Epoch 4139/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.9481e-04
Epoch 4140/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.5355e-04
Epoch 4141/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.8784e-04
Epoch 4142/6000
44/44 [==============================] - 1s 12ms/step - loss: 7.5953e-04
Epoch 4143/6000
44/44 [==============================] - 1s 12ms/step - loss: 8.7058e-04
Epoch 4144/6000
44/44 [====================

44/44 [==============================] - 0s 9ms/step - loss: 8.3794e-04
Epoch 4226/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.4362e-04
Epoch 4227/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.9864e-04
Epoch 4228/6000
44/44 [==============================] - 0s 10ms/step - loss: 7.8202e-04
Epoch 4229/6000
44/44 [==============================] - 0s 10ms/step - loss: 7.7168e-04
Epoch 4230/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.2665e-04
Epoch 4231/6000
44/44 [==============================] - 1s 12ms/step - loss: 7.5665e-04
Epoch 4232/6000
44/44 [==============================] - 0s 10ms/step - loss: 8.2827e-04
Epoch 4233/6000
44/44 [==============================] - 0s 10ms/step - loss: 7.8761e-04
Epoch 4234/6000
44/44 [==============================] - 1s 13ms/step - loss: 8.3544e-04
Epoch 4235/6000
44/44 [==============================] - 1s 12ms/step - loss: 7.5439e-04
Epoch 4236/6000
44/44 [==================

44/44 [==============================] - 0s 10ms/step - loss: 8.2138e-04
Epoch 4319/6000
44/44 [==============================] - 0s 11ms/step - loss: 8.1941e-04
Epoch 4320/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.0236e-04
Epoch 4321/6000
44/44 [==============================] - 0s 11ms/step - loss: 7.6633e-04
Epoch 4322/6000
44/44 [==============================] - 0s 11ms/step - loss: 7.9603e-04
Epoch 4323/6000
44/44 [==============================] - 0s 11ms/step - loss: 8.1841e-04
Epoch 4324/6000
44/44 [==============================] - 0s 10ms/step - loss: 7.6503e-04
Epoch 4325/6000
44/44 [==============================] - 0s 11ms/step - loss: 7.9293e-04
Epoch 4326/6000
44/44 [==============================] - 0s 11ms/step - loss: 8.3027e-04
Epoch 4327/6000
44/44 [==============================] - 0s 10ms/step - loss: 9.1033e-04
Epoch 4328/6000
44/44 [==============================] - 0s 10ms/step - loss: 7.3940e-04
Epoch 4329/6000
44/44 [===============

44/44 [==============================] - 0s 9ms/step - loss: 7.4933e-04
Epoch 4411/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.7080e-04
Epoch 4412/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.5105e-04
Epoch 4413/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.8435e-04
Epoch 4414/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.2001e-04
Epoch 4415/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.2405e-04
Epoch 4416/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.6965e-04
Epoch 4417/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.5448e-04
Epoch 4418/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.7554e-04
Epoch 4419/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.2605e-04
Epoch 4420/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.6839e-04
Epoch 4421/6000
44/44 [=========================

44/44 [==============================] - 0s 10ms/step - loss: 7.6043e-04
Epoch 4503/6000
44/44 [==============================] - 0s 10ms/step - loss: 7.6926e-04
Epoch 4504/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.5635e-04
Epoch 4505/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.8612e-04
Epoch 4506/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.5807e-04
Epoch 4507/6000
44/44 [==============================] - 0s 11ms/step - loss: 7.4990e-04
Epoch 4508/6000
44/44 [==============================] - 0s 10ms/step - loss: 7.9134e-04
Epoch 4509/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.9023e-04
Epoch 4510/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.3805e-04
Epoch 4511/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.4074e-04
Epoch 4512/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.9790e-04
Epoch 4513/6000
44/44 [=====================

44/44 [==============================] - 0s 9ms/step - loss: 7.2331e-04
Epoch 4596/6000
44/44 [==============================] - 0s 9ms/step - loss: 8.1588e-04
Epoch 4597/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.8180e-04
Epoch 4598/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.6228e-04
Epoch 4599/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.7485e-04
Epoch 4600/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.5671e-04
Epoch 4601/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.9837e-04
Epoch 4602/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.6500e-04
Epoch 4603/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.8768e-04
Epoch 4604/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.4496e-04
Epoch 4605/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.8202e-04
Epoch 4606/6000
44/44 [=========================

44/44 [==============================] - 0s 9ms/step - loss: 7.3816e-04
Epoch 4689/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.4446e-04
Epoch 4690/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.2774e-04
Epoch 4691/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.9025e-04
Epoch 4692/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.5303e-04
Epoch 4693/6000
44/44 [==============================] - 0s 8ms/step - loss: 7.7712e-04
Epoch 4694/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.8096e-04
Epoch 4695/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.9096e-04
Epoch 4696/6000
44/44 [==============================] - 0s 8ms/step - loss: 7.3001e-04
Epoch 4697/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.9823e-04
Epoch 4698/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.3855e-04
Epoch 4699/6000
44/44 [=========================

Epoch 4781/6000
44/44 [==============================] - 0s 10ms/step - loss: 7.5344e-04
Epoch 4782/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.2733e-04
Epoch 4783/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.2596e-04
Epoch 4784/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.7615e-04
Epoch 4785/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.4155e-04
Epoch 4786/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.6797e-04
Epoch 4787/6000
44/44 [==============================] - 0s 10ms/step - loss: 7.0905e-04
Epoch 4788/6000
44/44 [==============================] - 0s 10ms/step - loss: 7.6341e-04
Epoch 4789/6000
44/44 [==============================] - 0s 10ms/step - loss: 7.8304e-04
Epoch 4790/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.4443e-04
Epoch 4791/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.9034e-04
Epoch 4792/6000
44/44 [=====

44/44 [==============================] - 0s 9ms/step - loss: 7.4165e-04
Epoch 4874/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.4184e-04
Epoch 4875/6000
44/44 [==============================] - 0s 8ms/step - loss: 7.1170e-04
Epoch 4876/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.2344e-04
Epoch 4877/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.9035e-04
Epoch 4878/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.8763e-04
Epoch 4879/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.2287e-04
Epoch 4880/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.3602e-04
Epoch 4881/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.5128e-04
Epoch 4882/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.8288e-04
Epoch 4883/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.2671e-04
Epoch 4884/6000
44/44 [=========================

44/44 [==============================] - 0s 10ms/step - loss: 7.2801e-04
Epoch 4967/6000
44/44 [==============================] - 0s 10ms/step - loss: 7.5994e-04
Epoch 4968/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.1346e-04
Epoch 4969/6000
44/44 [==============================] - 0s 10ms/step - loss: 7.2510e-04
Epoch 4970/6000
44/44 [==============================] - 0s 10ms/step - loss: 7.7082e-04
Epoch 4971/6000
44/44 [==============================] - 0s 10ms/step - loss: 6.7447e-04
Epoch 4972/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.5141e-04
Epoch 4973/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.2518e-04
Epoch 4974/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.0653e-04
Epoch 4975/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.2260e-04
Epoch 4976/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.4134e-04
Epoch 4977/6000
44/44 [====================

Epoch 5059/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.2100e-04
Epoch 5060/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.4344e-04
Epoch 5061/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.2857e-04
Epoch 5062/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.6711e-04
Epoch 5063/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.9892e-04
Epoch 5064/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.6225e-04
Epoch 5065/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.2398e-04
Epoch 5066/6000
44/44 [==============================] - 0s 10ms/step - loss: 7.7656e-04
Epoch 5067/6000
44/44 [==============================] - 0s 10ms/step - loss: 6.7894e-04
Epoch 5068/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.4315e-04
Epoch 5069/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.3697e-04
Epoch 5070/6000
44/44 [=======

44/44 [==============================] - 0s 10ms/step - loss: 7.7865e-04
Epoch 5152/6000
44/44 [==============================] - 0s 11ms/step - loss: 6.4475e-04
Epoch 5153/6000
44/44 [==============================] - 0s 10ms/step - loss: 6.4577e-04
Epoch 5154/6000
44/44 [==============================] - 1s 11ms/step - loss: 7.3347e-04
Epoch 5155/6000
44/44 [==============================] - 1s 12ms/step - loss: 7.6507e-04
Epoch 5156/6000
44/44 [==============================] - 0s 10ms/step - loss: 6.4643e-04
Epoch 5157/6000
44/44 [==============================] - 0s 11ms/step - loss: 6.8507e-04
Epoch 5158/6000
44/44 [==============================] - 0s 10ms/step - loss: 6.7803e-04
Epoch 5159/6000
44/44 [==============================] - 0s 10ms/step - loss: 6.7949e-04
Epoch 5160/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.0434e-04
Epoch 5161/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.1563e-04
Epoch 5162/6000
44/44 [================

44/44 [==============================] - 0s 9ms/step - loss: 7.1659e-04
Epoch 5244/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.1797e-04
Epoch 5245/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.1881e-04
Epoch 5246/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.7073e-04
Epoch 5247/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.9359e-04
Epoch 5248/6000
44/44 [==============================] - 0s 10ms/step - loss: 6.6844e-04
Epoch 5249/6000
44/44 [==============================] - 0s 11ms/step - loss: 7.1006e-04
Epoch 5250/6000
44/44 [==============================] - 0s 10ms/step - loss: 7.0337e-04
Epoch 5251/6000
44/44 [==============================] - 0s 10ms/step - loss: 6.9950e-04
Epoch 5252/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.6955e-04
Epoch 5253/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.1671e-04
Epoch 5254/6000
44/44 [=====================

44/44 [==============================] - 0s 9ms/step - loss: 6.0459e-04
Epoch 5336/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.3154e-04
Epoch 5337/6000
44/44 [==============================] - 0s 10ms/step - loss: 6.3879e-04
Epoch 5338/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.1384e-04
Epoch 5339/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.8121e-04
Epoch 5340/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.0996e-04
Epoch 5341/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.2175e-04
Epoch 5342/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.9934e-04
Epoch 5343/6000
44/44 [==============================] - 0s 10ms/step - loss: 6.9414e-04
Epoch 5344/6000
44/44 [==============================] - 0s 10ms/step - loss: 7.0461e-04
Epoch 5345/6000
44/44 [==============================] - 0s 11ms/step - loss: 6.3897e-04
Epoch 5346/6000
44/44 [=====================

44/44 [==============================] - 0s 10ms/step - loss: 6.1208e-04
Epoch 5428/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.9694e-04
Epoch 5429/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.7599e-04
Epoch 5430/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.7262e-04
Epoch 5431/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.3466e-04
Epoch 5432/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.2476e-04
Epoch 5433/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.4793e-04
Epoch 5434/6000
44/44 [==============================] - 0s 10ms/step - loss: 6.8274e-04
Epoch 5435/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.8684e-04
Epoch 5436/6000
44/44 [==============================] - 0s 10ms/step - loss: 6.4521e-04
Epoch 5437/6000
44/44 [==============================] - 1s 12ms/step - loss: 7.3027e-04
Epoch 5438/6000
44/44 [=====================

44/44 [==============================] - 0s 9ms/step - loss: 7.2271e-04
Epoch 5520/6000
44/44 [==============================] - 0s 9ms/step - loss: 5.9944e-04
Epoch 5521/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.7490e-04
Epoch 5522/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.5277e-04
Epoch 5523/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.6396e-04
Epoch 5524/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.5551e-04
Epoch 5525/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.3907e-04
Epoch 5526/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.7295e-04
Epoch 5527/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.7586e-04
Epoch 5528/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.1940e-04
Epoch 5529/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.6716e-04
Epoch 5530/6000
44/44 [=========================

Epoch 5612/6000
44/44 [==============================] - 0s 10ms/step - loss: 6.3346e-04
Epoch 5613/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.6016e-04
Epoch 5614/6000
44/44 [==============================] - 1s 17ms/step - loss: 6.1991e-04
Epoch 5615/6000
44/44 [==============================] - 1s 18ms/step - loss: 7.4557e-04
Epoch 5616/6000
44/44 [==============================] - 1s 12ms/step - loss: 6.8124e-04
Epoch 5617/6000
44/44 [==============================] - 1s 12ms/step - loss: 6.2213e-04
Epoch 5618/6000
44/44 [==============================] - 1s 11ms/step - loss: 6.8855e-04
Epoch 5619/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.6020e-04
Epoch 5620/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.6263e-04
Epoch 5621/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.8829e-04
Epoch 5622/6000
44/44 [==============================] - 1s 14ms/step - loss: 6.8346e-04
Epoch 5623/6000
44/44 [==

Epoch 5705/6000
44/44 [==============================] - 0s 8ms/step - loss: 5.9072e-04
Epoch 5706/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.0091e-04
Epoch 5707/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.8596e-04
Epoch 5708/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.7932e-04
Epoch 5709/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.3321e-04
Epoch 5710/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.5976e-04
Epoch 5711/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.8760e-04
Epoch 5712/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.0830e-04
Epoch 5713/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.3096e-04
Epoch 5714/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.4826e-04
Epoch 5715/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.8847e-04
Epoch 5716/6000
44/44 [=========

44/44 [==============================] - 0s 10ms/step - loss: 6.6527e-04
Epoch 5799/6000
44/44 [==============================] - 0s 10ms/step - loss: 6.5784e-04
Epoch 5800/6000
44/44 [==============================] - 0s 10ms/step - loss: 6.3988e-04
Epoch 5801/6000
44/44 [==============================] - 0s 10ms/step - loss: 6.0342e-04
Epoch 5802/6000
44/44 [==============================] - 0s 10ms/step - loss: 6.6100e-04
Epoch 5803/6000
44/44 [==============================] - 0s 11ms/step - loss: 6.9587e-04
Epoch 5804/6000
44/44 [==============================] - 0s 10ms/step - loss: 6.1739e-04
Epoch 5805/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.1179e-04
Epoch 5806/6000
44/44 [==============================] - 0s 8ms/step - loss: 6.7567e-04
Epoch 5807/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.4649e-04
Epoch 5808/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.6544e-04
Epoch 5809/6000
44/44 [==================

44/44 [==============================] - 0s 9ms/step - loss: 6.1514e-04
Epoch 5892/6000
44/44 [==============================] - 0s 8ms/step - loss: 6.5742e-04
Epoch 5893/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.1928e-04
Epoch 5894/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.1721e-04
Epoch 5895/6000
44/44 [==============================] - 0s 8ms/step - loss: 6.0679e-04
Epoch 5896/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.9850e-04
Epoch 5897/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.3554e-04
Epoch 5898/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.1833e-04
Epoch 5899/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.6342e-04
Epoch 5900/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.0395e-04
Epoch 5901/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.5469e-04
Epoch 5902/6000
44/44 [=========================

44/44 [==============================] - 0s 9ms/step - loss: 6.4951e-04
Epoch 5984/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.6446e-04
Epoch 5985/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.0027e-04
Epoch 5986/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.3827e-04
Epoch 5987/6000
44/44 [==============================] - 0s 9ms/step - loss: 5.9757e-04
Epoch 5988/6000
44/44 [==============================] - 0s 9ms/step - loss: 7.1210e-04
Epoch 5989/6000
44/44 [==============================] - 0s 9ms/step - loss: 5.9256e-04
Epoch 5990/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.7618e-04
Epoch 5991/6000
44/44 [==============================] - 0s 9ms/step - loss: 6.9381e-04
Epoch 5992/6000
44/44 [==============================] - 0s 10ms/step - loss: 6.1220e-04
Epoch 5993/6000
44/44 [==============================] - 0s 10ms/step - loss: 6.6655e-04
Epoch 5994/6000
44/44 [=======================

In [92]:
pred = pd.DataFrame(model.predict(X_test)).set_index(y_test.index)
pred.columns = ["GOOG (Pred)"]
pred_df = pd.concat([y_test, pred], axis = 1)

2022-01-23 18:08:51.911755: I tensorflow/core/grappler/optimizers/custom_graph_optimizer_registry.cc:112] Plugin optimizer for device_type GPU is enabled.


In [94]:
plot_df = pd.concat([X_test,pred_df], axis = 1)

fig = px.line(plot_df)

fig.update_layout(legend_title = "Asset", title = "Asset Prices over Time ('GOOG' Predicted with Deep Learning)")
fig.update_xaxes(title = "Date")
fig.update_yaxes(title = "Stock Prices (in USD)")
fig.update_layout(width = 950, height = 700, title_x = 0.5, template = "ggplot2")

fig.write_image('Images/Asset_Prices_Predicted_Deep.png')

![title](Images/Asset_Prices_Predicted_Deep.png)

Here we can see that our model does not perform *tooooo* well. It seems to underestimate the price. That is better than overestimating since that could potentially lead to heavy losses. However, the model could be improved by adding layers and epochs! For now, this is more or less only a proof of concept.